### DISCUSSION TITLE
Detecting network failures / slow connections

### DISCUSSION THREAD

**Discussion by luizpvas** [livewire/livewire#3460] [1. Help]
Hello everyone :cat: 

I've been playing with Livewire over the weekend and I've been loving the experience. I have a few questions about how to detect network issues, request timeout and re-attempts.

1. Is there a local (e.g. in the button that triggered the action with wire:click) and/or a global way (e.g. from a browser event to show a toast alert) to detect when a remote action fails? It could be due to an exception or network issue.
2. It seems that `wire:loading` gets hidden when the action fails because the user is offline, is that right?
3. Is there a way to set a timeout for action requests?
4. Would it be possible to hook into livewire and implement something similar to [axios-retry](https://www.npmjs.com/package/axios-retry)? I know it would be dangerous for some actions, but most GET filtering/navigation would be fine.

A lot of my customers work from mobile phone and have poor internet connection, that's why I'm always worried about things like these :sweat_smile: 

If anyone could point me in the right direction it would be awesome. Thanks!

**Comment by jeffchown** [comment]
Hey @luizpvas, maybe the docs re offline state will help as a starting point - https://laravel-livewire.com/docs/2.x/offline-state
